src/components/Subthought.tsx

```diff
@@ -115,6 +117,18 @@ const Subthought = ({
     ref.current.style.opacity = opacity
   })
 
+  // Capture the static HTML string when the thought is first rendered
+  useEffect(() => {
+    if (thought && ref.current) {
+      cachedHTMLRef.current = ref.current.innerHTML
+    }
+  }, [thought])
+
+  // If the thought is deleted, return the cached static HTML from the ref
+  if (!thought && cachedHTMLRef.current) {
+    return <div dangerouslySetInnerHTML={{ __html: cachedHTMLRef.current }} />
```

Comment:
Personally, I'm generally okay with carefully using `dangerouslySetInnerHTML`. In this case, I don't see much risk related to e.g. XSS vulnerabilities since both the input and rendered HTML come from the user.

It could _maybe_ be argued that a user could be tricked into pasting and then deleting a thought with a malicious `<script>` in it, but we seem to be escaping `<` and `>` in thoughts regardless.

Just to be sure, have you considered if there are other options?

Reply:
@trevinhofmann I've tried a few different approaches before including:
- Storing thought directly to `useRef`, but it was ineffective since `Subthought` has a short-circuit (see below).
- Removing the short-circuit in `Subthought`, turned out to be too complicated since the `Thought` component still relies on the redux store.

Reply:
I have fixed the missing frame. This came from `editable--preventAutoscroll_true` class that animates the editable element's opacity. Since it's just a cache and it should always be visible, I perform a cleanup for that class before rendering it to the cached HTML.

I've also removed the `isLastActionDeleteThought` check so that it applies to the collapsing of context too.